### DISCUSSION TITLE
oracledb.exceptions.OperationalError: DPY-6000 thin mode

### DISCUSSION THREAD

**Discussion by vickyeu03** [oracle/python-oracledb#58] [Q&A]
Hi, I've been trying to connect to my oracle db with no luck. 
I know connection strings are fine since if I use think mode it works, but I need to use thin one. 

Im getting oracledb.exceptions.OperationalError: DPY-6000: cannot connect to database. Listener refused connection. (Similar to ORA-12660). 

Any help is appreciated.

**Comment by anthony-tuininga** [comment]
This is the same as #47. This is a known limitation. You can adjust the database configuration (if that is possible for you) or you will have to use thick mode for now!

**Reply by vickyeu03** [reply]
is there more explanation about how to adjust the database configuration? using thick mode is not an option in aws

**Reply by anthony-tuininga** [reply]
Discussion #50 has more detail. Essentially you have to comment out the lines in the configuration file sqlnet.ora on the server so that the encryption and inegrity services are not marked as requuired -- since these are not currently supported by the thin driver.

**Reply by anthony-tuininga** [reply]
The most important information for you is near the end of the discussion.

**Reply by cjbj** [reply]
Our general recommendation is, of course, to keep encryption enabled.  If you are using NNE, and cannot switch to TLS, this means you need to use Thick mode of python-oracledb

**Reply by vickyeu03** [reply]
In case I need to use Thick mode, using docker in aws. is there any tutorial? I could not install the client or access it.

**Reply by cjbj** [reply]
There are various resources for cx_Oracle (or SQL*Plus) and AWS that Google pops up for me.   They look useful.
